eng/targets/ResolveReferences.targets

```diff
@@ -148,6 +148,68 @@
+      BeforeTargets="AssignProjectConfiguration;
+          GenerateBuildDependencyFile;
+          GeneratePublishDependencyFile;
+          ILLink;
+          ResolveLockFileCopyLocalFiles;
+          ResolveLockFileReferences"
+      DependsOnTargets="ResolvePackageAssets">
+    <!-- Find projects patched in this build that need to be added as references. -->
+    <ItemGroup>
+      <_ProjectsInPatch Remove="@(_ProjectsInPatch)" />
+      <_ProjectsInPatch Include="@(ProjectReferenceProvider)"
+          Condition=" $(PackagesInPatch.Contains(' %(Identity);')) " />
+      <!-- dotnet-razorpagegenerator project is unique in overriding its package ID. -->
+      <_ProjectsInPatch Include="@(ProjectReferenceProvider->WithMetadataValue('Identity', 'dotnet-razorpagegenerator'))"
+          Condition=" $(PackagesInPatch.Contains(' RazorPageGenerator;')) " />
+
+      <!-- Ignore unreferenced projects/assemblies. Project reference providers have matching assembly and project names. -->
+      <_ProjectsToAdd Remove="@(_ProjectsToAdd)" />
+      <_ProjectsToAdd Include="@(_ProjectsInPatch)"
+          Condition=" @(RuntimeCopyLocalItems->WithMetadataValue('Filename', '%(_ProjectsToAdd.Identity)')->Count()) != 0 OR
+               @(ResolvedCompileFileDefinitions->WithMetadataValue('Filename', '%(_ProjectsToAdd.Identity)')->Count()) != 0 " />
+    </ItemGroup>
+
+    <Message Importance="high" Text="RuntimeCopyLocalItems: &apos;@(RuntimeCopyLocalItems->'%(Filename)')&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
+    <Message Importance="high" Text="RuntimeCopyLocalItems / 'Microsoft.AspNetCore.Http': &apos;@(RuntimeCopyLocalItems->WithMetadataValue('Filename', 'Microsoft.AspNetCore.Http'))&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
+    <Message Importance="high" Text="RuntimeCopyLocalItems / %(_ProjectsInPatch.Identity): &apos;@(RuntimeCopyLocalItems->WithMetadataValue('Filename', '%(_ProjectsInPatch.Identity)'))&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
+    <Message Importance="high" Text="ResolvedCompileFileDefinitions: &apos;@(ResolvedCompileFileDefinitions->'%(Filename)')&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
+    <Message Importance="high" Text="ResolvedCompileFileDefinitions / 'Microsoft.AspNetCore.Http': &apos;@(ResolvedCompileFileDefinitions->WithMetadataValue('Filename', 'Microsoft.AspNetCore.Http'))&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
+    <Message Importance="high" Text="ResolvedCompileFileDefinitions / %(_ProjectsInPatch.Identity): &apos;@(ResolvedCompileFileDefinitions->WithMetadataValue('Filename', '%(_ProjectsInPatch.Identity)'))&apos;"
+        Condition=" '$(MSBuildProjectName)' == 'ApplicationInsightsHostingStartupSample' " />
```

Comment:
These tasks output the following, indicating the Microsoft.AspNetCore.Http filename exists in both groups but isn't found when using `%(_ProjectsInPatch.Identity)`:
``` text
...
  lection;System.Reflection.Metadata;System.Reflection.Primitives;System.Reflection.TypeExtensions;System.Resources.Res
  ourceManager;System.Runtime;System.Runtime.CompilerServices.Unsafe;System.Runtime.Extensions;System.Runtime.Handles;S
  ystem.Runtime.InteropServices;System.Runtime.InteropServices.RuntimeInformation;System.Runtime.Numerics;System.Securi
  ty.AccessControl;System.Security.Cryptography.Algorithms;System.Security.Cryptography.Cng;System.Security.Cryptograph
  y.Encoding;System.Security.Cryptography.Primitives;System.Security.Cryptography.X509Certificates;System.Security.Cryp
  tography.Xml;System.Security.Permissions;System.Security.Principal.Windows;System.Text.Encoding;System.Text.Encoding.
  Extensions;System.Text.Encodings.Web;System.Threading;System.Threading.Tasks;System.Threading.Tasks.Parallel;System.V
  alueTuple;System.Xml.ReaderWriter;System.Xml.XDocument'
  ResolvedCompileFileDefinitions / 'Microsoft.AspNetCore.Http': 'C:\Users\dougbu\.nuget\packages\microsoft.aspnetcore.h
  ttp\2.1.1\lib\netstandard2.0\Microsoft.AspNetCore.Http.dll'
  ResolvedCompileFileDefinitions / Microsoft.AspNetCore.DataProtection.AzureKeyVault: ''
  ResolvedCompileFileDefinitions / Microsoft.AspNetCore.DataProtection.AzureStorage: ''
  ResolvedCompileFileDefinitions / Microsoft.AspNetCore.Http: ''
```

Output for the `@(RuntimeCopyLocalItems)` item group is similar.